lightningd/hsm_control.c

```diff
@@ -146,6 +147,43 @@ static struct command_result *json_getsharedsecret(struct command *cmd,
 	return command_success(cmd, response);
 }
 
+static struct command_result *json_getsecret(struct command *cmd,
```

Comment:
Is this not already achieved by `getsharedsecret`? I'm always a bit hesitant to add generic "give me a secret" methods as they may lead to extracting secrets we don't want to give away.

Reply:
`getsharedsecret` needs a compressed public key and then it performs ecdh() to give out a key for connection handshake and for onion wrapping.

We needed a pseudorandom key derived from hsm_secret, using hkdf_sha256() taking in a string as info, but I get your point. 
what if instead of taking in an 'info' string from the user I hard code it to "peer backup" to get a specific key used to encrypt the scb file?

Reply:
Ummmm just generate a random compressed public key?  Pick one out of a hat, then call `getsharedsecret` and then use the result as the encryption key, AND store the compressed public key in the first 33 bytes of the emergency-recovery file.  This lets you recover it anyway with the same `hsm_secret`; you only need the session pubkey.

Reply:
Secondarily, the VLS people have implemented a hardware version of the `hsmd`, and adding a new API to the `hsmd` requires that we at least inform them so theey can implement as well.  So I would be reluctant to add a new `hsmd_derive_secret` message here, we should try to use `getsharedsecret` and just give a random point that you store with the backup file anyway.  The random point will not leak the encryption key because of ECDH magic so it is safe to put it right beside the encrypted blob in the backup file.